src/service/exception_handlers.py

```diff
@@ -0,0 +1,70 @@
+"""
+Exception handlers for the FastAPI application.
+"""
+
+import logging
+
+from fastapi import HTTPException, Request, status
+from fastapi.responses import JSONResponse
+
+from src.service.error_mapping import map_error
+from src.service.exceptions import SparkManagerError
+
+logger = logging.getLogger(__name__)
+
+
+def _format_error(
+    status_code: int, error_code: int, error_type_str: str, message: str | None
+):
+    """Format error response with consistent structure."""
+    return JSONResponse(
+        status_code=status_code,
+        content={
+            "error": error_code,
+            "error_type": error_type_str,
+            "message": message or error_type_str,
+        },
+    )
+
+
+async def universal_error_handler(request: Request, exc: Exception):
+    """
+    Universal handler for all types of exceptions.
+
+    Handles:
+    - SparkManagerError and its subclasses (including all specific error types)
+    - HTTPException from FastAPI
+    - Generic exceptions
+    """
+    # Default values
+    error_code = 0
```

Comment:
None would be less confusing than 0 IMO

Reply:
I am fine with it rather than fixing all the type errors since it can be None and int.

Reply:
A clear api is more important than appeasing a clunky typing system

Reply:
I don’t see why None is clearer than 0. It’s the unnecessary complexity we should avoid.

Reply:
Because None clearly says there's no error code here, where 0 suggests there an error code 0, but there isn't. It really means there's no error code.

Reply:
👍 okay.